frontend/lib/pages/HomePage.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:flutter/material.dart';
+import 'package:frontend/pages/profilePage.dart';
+import 'package:frontend/pages/settingsPage.dart';
+import 'package:frontend/widgets/customInput.dart';
+import 'package:frontend/widgets/header.dart';
+import 'package:frontend/widgets/sidebar.dart';
+
+import '../widgets/customButton.dart';
+
+import './foodItemPage.dart';
+
+class HomePage extends StatefulWidget {
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  final emailInput = TextEditingController();
+  final passwordInput = TextEditingController();
+  bool isAuth = false;
+  PageController pageController;
+  int pageIndex = 0;
+  var currentPage = ["MealCare", "Profile", "Settings"];
+  var assetImage = new AssetImage("assets/logo.png");
+
+  @override
+  void initState() {
+    super.initState();
+    pageController = PageController();
+  }
+
+  void login(username, password) {
+    // here goes the function for login button
+    print("login Pressed");
+    print(username + " " + password);
+    setState(() {
+      isAuth = true;
+    });
+  }
+
+  void logout() {
+    setState(() {
+      isAuth = false;
+    });
+  }
+
+  @override
+  void dispose() {
+    pageController.dispose();
+    super.dispose();
+  }
+
+  onPageChanged(int pageIndex) {
+    setState(() {
+      this.pageIndex = pageIndex;
+    });
+  }
+
+  changePage(int pageIndex) {
+    pageController.animateToPage(pageIndex,
+        duration: Duration(milliseconds: 200), curve: Curves.ease);
+    setState(() {
+      this.pageIndex = pageIndex;
+    });
+  }
+
+  Widget buildAuthScreen() {
+    var _ScaffoldKey = new GlobalKey<ScaffoldState>();
+    return Scaffold(
+      key: _ScaffoldKey,
+      appBar: Header(
+          isAppTitle: true, text: currentPage[pageIndex], s: _ScaffoldKey),
+      body: PageView(
+        physics: NeverScrollableScrollPhysics(),
+        controller: pageController,
+        children: <Widget>[
```

Comment:
I like the idea of the page view, but I feel like the navigation using a router would suit the UI more and is easier to follow than the changePage(int) function which could be hard to read.

Reply:
Ya but don't worry about this function. Now we can work separately on the design and the functionalities of the pages rather than worrying about routes and authentication each time.

Reply:
I personally think we should worry because this changes the whole app structure it's something we should definitely have a meeting on

Reply:
Ya sure, i'm open for meeting. pretty much free all day lol